lib/mayaUsd/render/vp2RenderDelegate/proxyRenderDelegate.cpp

```diff
@@ -823,24 +823,16 @@ void ProxyRenderDelegate::_UpdateRenderTags()
         }
     }
 
-    // When the render tag on an rprim changes we do a pass over all rprims to update
-    // their visibility. The frame after we do the pass over all the tags, set the tags back to
-    // the minimum set of tags.
-    if (!_taskRenderTagsValid) {
-        TfTokenVector renderTags
-            = { HdRenderTagTokens->geometry }; // always draw geometry render tag purpose.
-        if (_proxyShapeData->DrawRenderPurpose()) {
-            renderTags.push_back(HdRenderTagTokens->render);
-        }
-        if (_proxyShapeData->DrawProxyPurpose()) {
-            renderTags.push_back(HdRenderTagTokens->proxy);
-        }
-        if (_proxyShapeData->DrawGuidePurpose()) {
-            renderTags.push_back(HdRenderTagTokens->guide);
-        }
-        _taskController->SetRenderTags(renderTags);
-        _taskRenderTagsValid = true;
-    }
+    // The renderTagsVersion increments when the render tags on an rprim changes, or when the
+    // global render tags are set. Check to see if the render tags version has changed since
+    // the last time we set the render tags so we know if there is a change to an individual
+    // rprim or not.
+    bool rprimRenderTagChanged = (_renderTagVersion != changeTracker.GetRenderTagVersion());
+#if USD_VERSION_NUM <= 2005
+    // in v20.05 and earlier when the purpose of an rprim changes the visibility gets dirtied,
+    // and that doesn't update the render tag version.
+    rprimRenderTagChanged = rprimRenderTagChanged || ( _visibilityVersion != changeTracker.GetVisibilityChangeCount());
```

Comment:
Extra check necessary until purpose changing correctly marks DirtyRenderTag instead of DirtyVisibility.